extensions-core/namespace-lookup/src/test/java/io/druid/query/lookup/NamespaceLookupExtractorFactoryTest.java

```diff
@@ -0,0 +1,461 @@
+      @Override
+      public long getPollMs()
+      {
+        return 0;
+      }
+    };
+    EasyMock.expect(cacheManager.scheduleAndWait(
+        EasyMock.anyString(),
+        EasyMock.eq(extractionNamespace),
+        EasyMock.eq(60000L)
+    )).andReturn(false).once();
+    EasyMock.expect(
+        cacheManager.checkedDelete(EasyMock.anyString())
+    ).andReturn(false).once();
+    EasyMock.replay(cacheManager);
+
+    final NamespaceLookupExtractorFactory namespaceLookupExtractorFactory = new NamespaceLookupExtractorFactory(
+        extractionNamespace,
+        cacheManager
+    );
+    Assert.assertFalse(namespaceLookupExtractorFactory.start());
+    Assert.assertFalse(namespaceLookupExtractorFactory.close());
+    EasyMock.verify(cacheManager);
+  }
+
+  @Test
+  public void testReplaces()
+  {
+    final ExtractionNamespace en1 = EasyMock.createStrictMock(ExtractionNamespace.class), en2 = EasyMock.createStrictMock(
+        ExtractionNamespace.class);
+    EasyMock.replay(en1, en2);
+    final NamespaceLookupExtractorFactory f1 = new NamespaceLookupExtractorFactory(
+        en1,
+        cacheManager
+    ), f2 = new NamespaceLookupExtractorFactory(en2, cacheManager), f1b = new NamespaceLookupExtractorFactory(
+        en1,
+        cacheManager
+    );
+    Assert.assertTrue(f1.replaces(f2));
+    Assert.assertTrue(f2.replaces(f1));
+    Assert.assertFalse(f1.replaces(f1b));
+    Assert.assertFalse(f1b.replaces(f1));
+    Assert.assertFalse(f1.replaces(f1));
+    Assert.assertTrue(f1.replaces(EasyMock.createNiceMock(LookupExtractorFactory.class)));
+    EasyMock.verify(en1, en2);
+  }
+
+  @Test(expected = ISE.class)
+  public void testMustBeStarted()
+  {
+    final ExtractionNamespace extractionNamespace = new ExtractionNamespace()
+    {
+      @Override
+      public long getPollMs()
+      {
+        return 0;
+      }
+    };
+
+    final NamespaceLookupExtractorFactory namespaceLookupExtractorFactory = new NamespaceLookupExtractorFactory(
+        extractionNamespace,
+        cacheManager
+    );
+
+    namespaceLookupExtractorFactory.get();
+  }
+
+  // Note this does NOT catch problems with returning factories as failed in error messages.
+  @Test
+  public void testSerDe() throws Exception
+  {
+    final Injector injector = Initialization.makeInjectorWithModules(
+        GuiceInjectors.makeStartupInjector(),
+        ImmutableList.of(
+            new Module()
+            {
+              @Override
+              public void configure(Binder binder)
+              {
+                JsonConfigProvider.bindInstance(
+                    binder, Key.get(DruidNode.class, Self.class), new DruidNode("test-inject", null, null)
+                );
+              }
+            }
+        )
+    );
+    final ObjectMapper mapper = injector.getInstance(Key.get(ObjectMapper.class, Json.class));
+    mapper.registerSubtypes(NamespaceLookupExtractorFactory.class);
+    final LookupExtractorFactory factory = mapper.readValue(
```

Comment:
I added some more serde related tests, but I'm testing `replaces` instead of `equals`